### ISSUE TITLE
[Linux] Build failure

### ISSUE THREAD

**Issue by Shoegzer** [hikari-no-yume/touchHLE#346]
### Describe the feature or issue.

Failure occurs during building.  The following seems to be the issue:

```
error: failed to run custom build command for `touchHLE_dr_mp3_wrapper v0.2.2 (/home/testing/Desktop/touchHLE/src/audio/dr_mp3_wrapper)`

Caused by:
  process didn't exit successfully: `/home/testing/Desktop/touchHLE/target/release/build/touchHLE_dr_mp3_wrapper-c55d608394105136/build-script-build` (exit status: 1)
```

Test environment:
* OS: Linux Mint 22 (Ubuntu 24.04 derivative) / kernel 6.8
* CPU: Intel i7-10870h
* Compiler: gcc-14.0.1
* Commit tested: 9ef94a9d6494a09071ab0e1068fdac9bfb5ee302
* Log: [touchhle_build_log.txt](https://github.com/user-attachments/files/16488674/touchhle_build_log.txt)

### By submitting this issue, I certify that…

- [X] I understand that linking to, discussing or asking for pirated content is not permitted, and will lead to an immediate ban.
- [X] I understand that “Abandonware” and illegal copies of free games are also pirated content.

**Comment by hikari-no-yume** [comment]
Linux is not a supported platform for us. The issue here looks like it would be a one-line fix if you want to fix it.

**Comment by Shoegzer** [comment]
Thanks and sure, and I can even submit a PR for it to help others.  What's the fix?

**Comment by parkerlreed** [comment]
I changed https://github.com/touchHLE/touchHLE/blob/trunk/src/audio/dr_mp3_wrapper/lib.c#L15

to drmp3_uint64 and it compiled. Haven't been able to test yet

**Comment by parkerlreed** [comment]
Seems to work. I have audio in Super Monkey Ball

**Comment by hikari-no-yume** [comment]
Super Monkey Ball doesn't use MP3 fwiw. But the issue is a trivial `unsigned long` (64-bit) versus `unsigned long long` (also 64-bit) implicit conversion issue, I don't expect fixing it to break things.

**Comment by parkerlreed** [comment]
So is that the correct fix? I can throw up a PR if needed.

**Comment by hikari-no-yume** [comment]
The correct fix is to add an explicit cast.

**Comment by hikari-no-yume** [comment]
Fixed in 50d6673e9abad44e819a03c1dfb8eb6e0304a9b5.
